### PR TITLE
Shyrma multiply bp

### PR DESCRIPTION
**Pull request by shyrma**
refine multiply_ff and correct multiply_bp

### CODE REVIEW

File: libnd4j/include/ops/declarable/generic/broadcastable/multiply.cpp [hunk 1]
```diff
@@ -16,160 +16,158 @@
...
+    Nd4jLong *dLdxShapeInfo = nullptr;
+    Nd4jLong *dLdyShapeInfo = nullptr;
+
+    COPY_SHAPE(xShapeInfo, dLdxShapeInfo);
+    COPY_SHAPE(yShapeInfo, dLdyShapeInfo);
+
+    return SHAPELIST(dLdxShapeInfo, dLdyShapeInfo);     
+}
+
+        // CUSTOM_OP_IMPL(multiply_bp, 3, 2, false, 0, 0) {
```
Comment (raver119) [inline libnd4j/include/ops/declarable/generic/broadcastable/multiply.cpp]: Please do not do this. Use block comments instead.

File: libnd4j/include/ops/declarable/generic/broadcastable/multiply.cpp [hunk 2]
```diff
@@ -16,160 +16,158 @@
...
+    NDArray<T> *x = INPUT_VARIABLE(0);
+    NDArray<T> *y = INPUT_VARIABLE(1);
+    NDArray<T> *z = OUTPUT_VARIABLE(0);
+
+    Nd4jLong* zShapeInfo = nullptr;
+    const bool areShapesBroadcastable = ShapeUtils<T>::evalBroadcastShapeInfo(x->getShapeInfo(), y->getShapeInfo(), true, zShapeInfo, block.getWorkspace());
+    REQUIRE_TRUE(areShapesBroadcastable, 0, "MULTIPLY OP: the shapes of x %s and y %s are not suitable for broadcast !", ShapeUtils<T>::shapeAsString(x).c_str(), ShapeUtils<T>::shapeAsString(y).c_str());
+    RELEASE(zShapeInfo, block.getWorkspace());
+
+    z->assign(*x * *y);
```
Comment (raver119) [inline libnd4j/include/ops/declarable/generic/broadcastable/multiply.cpp]: Please get back original FF implementation that utilizes BroadcastHelper. No need inventing the wheel once again.

File: libnd4j/include/ops/declarable/generic/broadcastable/multiply.cpp (hunk 1 above)
Comment (shyrma) [reply libnd4j/include/ops/declarable/generic/broadcastable/multiply.cpp]: done

File: libnd4j/include/ops/declarable/generic/broadcastable/multiply.cpp (hunk 2 above)
Comment (shyrma) [reply libnd4j/include/ops/declarable/generic/broadcastable/multiply.cpp]: done

**Pull request merged**